addons/html_editor/static/src/core/clipboard_plugin.js

```diff
@@ -104,9 +104,15 @@ export class ClipboardPlugin extends Plugin {
     onCopy(ev) {
         ev.preventDefault();
         const selection = this.shared.getEditableSelection();
-
+        const commonAncestor = selection.commonAncestorContainer;
+        if (commonAncestor && commonAncestor.nodeType === Node.ELEMENT_NODE) {
+            this.dispatch("CLEAN", { root: commonAncestor });
+        }
```

Comment:
I don't know all the implications of dispatching `CLEAN` but shouldn't this be done on the clone instead? cc @cammarosano

Reply:
Yes, I think it would be better to do it on the clone.

Using CLEAN on the actual editable before cloning/copying something is the pattern used by the Split and Dom plugins. And that's what I based my initial suggestion on (https://github.com/odoo/odoo/pull/182596#issuecomment-2400061650). But now that you mentioned it... I'm not sure that's a good pattern at all.

And there's a problem I just realized now: even if NORMALIZE reverts the changes done by CLEAN, the added effect is a bunch of mutations that are not added to a step right away (as copying content adds no step), which is something we're trying to avoid.

So yes, I would suggest to:
- first clone the contents
- dispatch CLEAN on the clone
- no need to dispatch NORMALIZE anymore.

Just make sure the CLEAN handlers will not try to "preserve selection" (remove it from the editable and put it inside the clone) in this case.

Also, consider implementing CLEAN for the Format plugin (if we don't want FEFF in the copied content, I assume we don't want ZWS and [data-oe-zws-empty-inline] att either...).

Reply:
After some trials and failure I realized that the text content that we put in the clipboard here https://github.com/odoo/odoo/blob/d82522a0c5dbd5f13ed4d2852a697a587ae1f295/addons/html_editor/static/src/core/clipboard_plugin.js#L191
is actualy coming from the selection that we set here  https://github.com/odoo/odoo/blob/d82522a0c5dbd5f13ed4d2852a697a587ae1f295/addons/html_editor/static/src/core/selection_plugin.js#L261 and it's necessary for tests like this https://github.com/odoo/odoo/blob/d82522a0c5dbd5f13ed4d2852a697a587ae1f295/addons/html_editor/static/tests/copy.test.js#L34-L42 because using element.textContent or element.innerText doesn't change the `br` elements into `\n`, maybe there are other examples, but the conclusion is that we need to get the text content from a selection so it should be something in the dom, I have two ideas:
- We add the element with d-none and create a new range get the text and remove them after that
- We create a clean flow specific for strings which removes the ufeff and other stuff that may appear in the text content

cc @dmo-odoo , @cammarosano